cmd/flux/diff_artifact.go

```diff
@@ -17,66 +17,107 @@ limitations under the License.
 package main
 
 import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
 	"context"
+	"errors"
 	"fmt"
+	"io"
 	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
 
 	oci "github.com/fluxcd/pkg/oci/client"
 	sourcev1 "github.com/fluxcd/source-controller/api/v1beta2"
+	"github.com/gonvenience/ytbx"
+	"github.com/google/shlex"
+	"github.com/homeport/dyff/pkg/dyff"
 	"github.com/spf13/cobra"
+	"golang.org/x/exp/maps"
 
 	"github.com/fluxcd/flux2/v2/internal/flags"
+	"github.com/fluxcd/flux2/v2/pkg/printers"
 )
 
+var ErrDiffArtifactChanged = errors.New("the artifact contents differ")
+
 var diffArtifactCmd = &cobra.Command{
-	Use:   "artifact",
+	Use:   "artifact <from> <to>",
 	Short: "Diff Artifact",
-	Long:  withPreviewNote(`The diff artifact command computes the diff between the remote OCI artifact and a local directory or file`),
+	Long: withPreviewNote(fmt.Sprintf(
+		"The diff artifact command prints the diff between the remote OCI artifact and a local directory or file.\n\n"+
+			"You can overwrite the command used for diffing by setting the %q environment variable.", externalDiffVar)),
 	Example: `# Check if local files differ from remote
-flux diff artifact oci://ghcr.io/stefanprodan/manifests:podinfo:6.2.0 --path=./kustomize`,
```

Comment:
This is a breaking change, please undo it.

Reply:
The previous `flux diff artifact <path1> --path=<path2>` syntax is still supported for backwards compatibility. This simply makes `flux diff artifact <path1> <path2>` the _documented_ syntax. If you don't like it, I can revert back to the `--path` syntax, but compatibility should not be broken.